### ISSUE TITLE
Support systemd-resolved

### ISSUE THREAD

**Issue by fpauser** [kayrus/gof5#22]
It would be great if gof5 would support [`systemd-resolved`](https://fedoraproject.org/wiki/Changes/systemd-resolved#Documentation) (which is the new default in Fedora 33) for dns.

**Comment by kayrus** [comment]
@fpauser could you please clarify what exactly do you want to support? Currently gof5 works well on systems, where systemd-resolved is used.

**Comment by fpauser** [comment]
Hm, after updating to Fedora 33 which brings systemd-resolved by default dns stopped working when connected to the vpn - I had to disable systemd-resolved to get it working.

**Comment by kayrus** [comment]
@fpauser got it. Are you willing to introduce systemd-resolved support?

**Comment by kayrus** [comment]
I executed gof5 in fedora 33 and it works without additional modifications. Could you please specify which exact problems do you have?

**Comment by kayrus** [comment]
So, it is possible to avoid `/etc/resolv.conf` modification with a built-in DNS listener on systems, where systemd-resolved is installed. In addition it is also possible to avoid running gof5 with sudo. Short plan:
* check whether gof5 has CAP_NET_ADMIN capabilities
* check whether D-BUS is available and has systemd-resolved `/org/freedesktop/resolve1` node
* Set F5 DNS servers to the tun interface via DBUS resolved
* Set custom domains from the `cfg.DNS` list via DBUS resolved

some references for a future work:
* https://venam.nixers.net/blog/unix/2020/07/06/dbus-polkit.html
* https://github.com/godbus/dbus

**Comment by joker-at-work** [comment]
Bare in mind, that one can still choose to disable systemd-resolved by removing the entry in `/etc/resolv.conf`. Then I would assume it's still available via D-BUS.

**Comment by kayrus** [comment]
Unfortunately there is no proper way to run VPN client without sudo. It appeared that polkit-1 doesn't support capabilities and asks for a user password in every update action. Therefore in case when you need to set dns domains and dns servers, you need to enter your password twice. see https://github.com/systemd/systemd/issues/18956
@joker-at-work @fpauser is anyone interested on hacking godbus?

**Comment by kayrus** [comment]
Good news Everyone!
![GoodNewsEveryone](https://user-images.githubusercontent.com/4047483/110853454-f1a7e080-82b3-11eb-8812-7d45e903c34c.jpg)
I added a support for systemd-resolved. Try a new experimental branch https://github.com/kayrus/gof5/tree/resolved it has an updated [README](https://github.com/kayrus/gof5/blob/resolved/README.md#linux) as well.
Try it in your environment and let me know the test results.

**Comment by kayrus** [comment]
Should be fixed by cad4edb2f5fa2caf7b4f5f2a2b569e8c3261b69c

**Comment by Xyaren** [comment]
Question:
Shouldn't dns on systemd be supported without DNS proxy ?

```bash
 resolvectl dns tun0 "10.0.53.1" "10.0.53.2"
 resolvectl domain tun0 mycomp.net mycomp.insite
 ```
